scale/job/models.py

```diff
@@ -219,10 +219,12 @@ def get_details(self, job_id):
 
         # Attempt to get related job executions
         if job.status in ['RUNNING', 'COMPLETED', 'FAILED', 'QUEUED']:
-            job_exe = JobExecution.objects.filter(job=job, exe_num=job.num_exes).select_related('job', 'node', 'error')
-            job.execution = job_exe.defer('job__input', 'job__output').order_by('-created')
+            try:
+                job.execution = JobExecution.objects.get_job_exe_details(job_id=job.id, exe_num=job.num_exes)
+            except JobExecution.DoesNotExist():
```

Comment:
Last thing: extra parens here. Wait to merge until after the release is done. Well done.